### PR TITLE
add 'spec roadmap' section 

### PR DESCRIPTION
**Pull request by equalsJeffH**
After the end of the last webauthn F2F meeting, @vijaybh and I were discussing various spec editing stuff and how there's various disparate audiences for this spec (user agent developers, RP developers of various flavors, authenticator developers, etc), and also issue #301 re moving the example scenarios up to the front of the spec, came up. 

Anyway, I said I'd take a stab at concocting a "spec roadmap" section in order to give said audiences some guidance, and here's a very rough first cut.  Discussion/feedback appreciated.

improves #585

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/375.html" title="Last updated on Sep 12, 2018, 9:02 PM GMT (cf07d23)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/375/035ea79...cf07d23.html" title="Last updated on Sep 12, 2018, 9:02 PM GMT (cf07d23)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -96,6 +96,25 @@ device running the user agent, and be accessed over a transport such as
 Universal Serial Bus (USB), Bluetooth Low Energy (BLE) or Near Field
 Communications (NFC).
 
+## Specification Roadmap ## {#spec-roadmap}
+
+While many W3C specifications are directed primarily to audiences comprising user agent developers and web application developers (i.e., Web authors, herein also known as <em>[RP] developers</em>), the character of Web Authentication requires that this specification also be understood and properly used by those whose responsibilities include making decisions regarding:
```
Comment (vijaybh) [inline index.bs]: Nit: "also be understood" seems at odds with the repetition of "[RP] developers" below. Maybe say something like "... requires that this specification be understood and correctly used by multiple audiences, namely:"

File: index.bs [hunk 2]
```diff
@@ -96,6 +96,25 @@ device running the user agent, and be accessed over a transport such as
...
+
+- OS platform API design and implementation in regards to platform-specific [=authenticator=] APIs, platform [=WebAuthn Client=] instantiation, etc. (OS platform developers)
+
+- [=Authenticator=] developers
+
+User agent developers, web framework developers, and OS platform developers ought to thoroughly understand the entire specification, except for perhaps the details of the various attestation statement formats ([[#defined-attestation-formats]]), since the latter are blobs emitted by authenticators that simply pass-through platforms and user agents on their way to the [RP] server (unless, of course, their platform emits an attestation statement of its own).
+
+[RP] developers (web authors) will find the [[#use-cases]], [[#sample-scenarios]], and [[#terminology]] sections useful for providing overall context. Further, the [[#api]], [[#rp-operations]], and [[#extension-predef]] sections provide the details underlying the aforementioned sections. [RP] server-side developers ought to undertand [[#cred-attestation]] in general. They will need to understand the latter in particular, as well as [[#defined-attestation-formats]], if they intend to verify credential attestations.
```
Comment (vijaybh) [inline index.bs]: Typo: "undertand". Also should we add a sentence encouraging RPs to check attestation?

File: index.bs [hunk 3]
```diff
@@ -96,6 +96,25 @@ device running the user agent, and be accessed over a transport such as
...
+
+- [RP] web application login flows, account recovery flows, user account database content, etc. ([RP] developers)
+
+- Third-party web application frameworks / SDKs (Web framework developers)
+
+- OS platform API design and implementation in regards to platform-specific [=authenticator=] APIs, platform [=WebAuthn Client=] instantiation, etc. (OS platform developers)
+
+- [=Authenticator=] developers
+
+User agent developers, web framework developers, and OS platform developers ought to thoroughly understand the entire specification, except for perhaps the details of the various attestation statement formats ([[#defined-attestation-formats]]), since the latter are blobs emitted by authenticators that simply pass-through platforms and user agents on their way to the [RP] server (unless, of course, their platform emits an attestation statement of its own).
```
Comment (vijaybh) [inline index.bs]: Run-on sentence?

In general, it seems like each line here encourages each audience to understand the whole specification. In that case, breaking things down by audience seems unnecessary. Should we instead say something like "RP developers will want to start with the use cases and sample scenarios. Web authors will want to read the interface section carefully, and server developers will want to read the RP considerations section in detail... UA developers and web framework developers will want to start with the use cases and sample scenarios, and to read the interface section very carefully.... Authenticator developers will want to pay particular attention to the authenticator model and attestation sections..."? Also, could this be done in the respective bullet points above so we don't have two taxonomies of readers?

File: index.bs (hunk 1 above)
Comment (equalsJeffH) [reply index.bs]: done.

**Comment by equalsJeffH** [comment]
updated -- please re-review

File: index.bs [hunk 4]
```diff
@@ -705,12 +756,13 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
-    [=Registration=] and [=Authentication=], respectively.
+:: The entity whose <dfn>web application</dfn> utilizes the [[#api|Web Authentication API]] to [=registration|register=] and
+    [=authentication|authenticate=] users.
 
-        Note: While the term [=[RP]=] is used in other contexts (e.g., X.509 and OAuth), an entity acting as a [=[RP]=] in one
+        Note: While the term [=[RP]=] is also often used in other contexts (e.g., X.509 and OAuth), an entity acting as a [=[RP]=] in one
         context is not necessarily a [=[RP]=] in other contexts. In this specification, the term [=[WRP]=] is often shortened 
-        to be just [=[RP]=].
+        to be just [=[RP]=], and explicitly refers to a [=[RP]=] in the WebAuthn context. Note that in any concrete instantiation
+        that a WebAuthn context may be embedded in a broader overall context, e.g., one based on OAuth.
```
Comment (emlun) [inline index.bs]: I can't quite parse the last sentence - I think there might be one too many "that"s in it, or something similar.

File: index.bs [hunk 5]
```diff
@@ -257,6 +264,50 @@ for [=user verification=], along with appropriate driver software to mediate acc
...
+- [=Authenticator=] developers. These readers will want to pay particular attention to [[#sctn-authenticator-model]],
+    [[#defined-attestation-formats]], [[#extensions]], and [[#sctn-defined-extensions]].
+    <br><br>
+
+<div class="note">
+    Note: Along with the [[#api|Web Authentication API]] itself, this specification defines a 
+    request-response <em>cryptographic protocol</em>
+    between a [=[WRP]=] server and an [=authenticator=], comprised of 
+    {{PublicKeyCredentialCreationOptions|input data}} supplied by the [=[RP]=] and a digitally signed
+    [=authenticator data=] message emitted by the [=authenticator=] in reply. The latter protocol is in turn conveyed via the
```
Comment (emlun) [inline index.bs]: It's not clear to me what "the latter protocol" is meant to refer to. Perhaps this could be said as "This reply is in turn..."?

Comment (equalsJeffH) [reply index.bs]: it's referring to the cryptographic Authnr <--> RP protocol, described in the prior sentence.  I've re-written that parag to attempt to clarify.

File: index.bs (hunk 4 above)
Comment (equalsJeffH) [reply index.bs]: ok, 2nd "that" removed. better?

File: index.bs [hunk 6]
```diff
@@ -261,6 +268,55 @@ for [=user verification=], along with appropriate driver software to mediate acc
...
+    Note: Along with the [[#api|Web Authentication API]] itself, this specification defines a 
+    request-response <em>cryptographic protocol</em>
+    between a [=[WRP]=] server and an [=authenticator=], where the [=[RP]=]'s request consists of a
+    [[#cryptographic-challenges|challenge]] and other
+    input data supplied by the [=[RP]=] and sent to the [=authenticator=].
+    The request is conveyed via the
+    combination of HTTPS, the [=[RP]=] [=web application=], the [[#api|WebAuthn API]], and the platform-specific communications channel
+    between the user agent and the [=authenticator=].
+    The [=authenticator=] replies with a digitally signed [=authenticator data=] message and other output data, which is conveyed back to the
+    [=[RP]=] server via the same path (in reverse). Protocol details vary according to whether an [=authentication=] or
```
Comment (emlun) [inline index.bs]: Are the parentheses necessary?

Comment (emlun) [inline index.bs]: Are the parentheses necessary?

Comment (equalsJeffH) [reply index.bs]: done!

**Pull request merged**